uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -605,119 +541,61 @@ define([
       });
     });
 
-    describe('uploading files', function () {
-      var promise;
-
-      beforeEach(function () {
-        requestData = helper.createRandomLeaveRequest();
-        instance = LeaveRequestInstance.init(requestData);
-        spyOn(instance.fileUploader, 'uploadAll').and.callThrough();
-        instance.fileUploader.queue = [{ 'key': 2 }];
-      });
-
-      afterEach(function () {
-        $rootScope.$apply();
-      });
-
-      describe('on create()', function () {
-        beforeEach(function () {
-          promise = instance.create();
-        });
-
-        it('uploads file with entity id', function () {
-          promise.then(function () {
-            expect(instance.fileUploader.uploadAll).toHaveBeenCalledWith({ entityID: instance.id });
-          });
-        });
-      });
-
-      describe('on update()', function () {
-        beforeEach(function () {
-          instance.id = '12';
-          promise = instance.update();
-        });
-
-        it('uploads file with entity id', function () {
-          promise.then(function () {
-            expect(instance.fileUploader.uploadAll).toHaveBeenCalledWith({ entityID: instance.id });
-          });
-        });
-      });
-    });
-
     describe('attachments', function () {
-      var attachments, numberOfFiles, promise;
-      var testId = '63';
+      var attachments, instance, promise;
 
       beforeEach(function () {
-        LeaveRequestInstance.id = '12';
-        attachments = leaveRequestMockData.getAttachments().values;
-        numberOfFiles = attachments.length;
-        promise = LeaveRequestInstance.loadAttachments();
-      });
+        instance = LeaveRequestInstance.init(helper.createRandomLeaveRequest());
+        instance.id = '12';
 
-      afterEach(function () {
-        $rootScope.$apply();
+        attachments = leaveRequestMockData.getAttachments().values;
+        promise = instance.loadAttachments();
       });
 
       describe('loadAttachments()', function () {
-        it('initializes files array', function () {
+        it('fetches its own attachments', function () {
+          expect(LeaveRequestAPI.getAttachments).toHaveBeenCalledWith(instance.id);
+        });
+
+        it('stores the attachments internally', function () {
           promise.then(function () {
-            expect(LeaveRequestInstance.files.length).toEqual(numberOfFiles);
+            expect(instance.files.length).toEqual(attachments.length);
           });
         });
       });
 
       describe('deleteAttachment()', function () {
         beforeEach(function () {
-          promise.then(function () {
-            LeaveRequestInstance.deleteAttachment(attachments[0]);
-          });
+          $rootScope.$apply();
```

Comment:
no, this triggers the `loadAttachments()` in the `beforeEach` above. This would be the sequence
```
loadAttachments()
$apply() // triggers loadAttachments()
deleteAttachment()
```